File: apps/api/core/users-jobs-service/src/Domain/Entities/Developers/Developer.cs
namespace Domain.Entities.Developers;

using Attributes.Educations;
using Attributes.Experiences;
using Attributes.Languages;
using Attributes.Skills;
using Attributes.Specializations;
using Entities.Jobs;
using Enums;
using Joins.DeveloperLanguages;
using Joins.DeveloperSkills;
using Joins.JobDevelopers;
using SkDomain.Entities;
using Users;

public sealed class Developer : BaseEntity
{
    public required string Name { get; set; }
    public required string LastName { get; set; }
    public required double YearsOfExperience { get; set; }
    public required double SalaryPerHourExpected { get; set; }
    public required DeveloperModality ModalityPreferred { get; set; }

    public string? Location { get; set; }
    public Uri? ProfilePictureUrl { get; set; }
    public string? Description { get; set; }

    public required Guid UserId { get; init; }
    public User User { get; init; } = null!;

    public required Guid SpecializationId { get; init; }
    public Specialization Specialization { get; init; } = null!;

    public IList<Skill> Skills { get; } = [];
    public IList<DeveloperSkill> DeveloperSkills { get; } = [];

    public IList<Language> Languages { get; } = [];
    public IList<DeveloperLanguage> DeveloperLanguages { get; } = [];

    public IList<Job> Jobs { get; } = [];
    public IList<JobDeveloper> JobDevelopers { get; } = [];

    public IList<Experience> Experiences { get; } = [];
    public IList<Education> Educations { get; } = [];
}

File: apps/client/src/features/developer/types/developer.ts
export interface Developer {
  id: string;
  name: string;
  lastName: string;
  yearsOfExperience: number;
  salaryPerHourExpected: number;
  location: string;
  portfolioUrl: string;
  profilePictureUrl: string;
  description: string;
  specialization: {
    id: string;
    name: string;
  };
  skills: Array<{
    id: string;
    name: string;
  }>;
  spokenLanguages: Array<{
    id: string;
    name: string;
  }>;
}

export interface PaginatedResponse<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalCount: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export interface DeveloperFilters {
  specialization?: string;
  minExperience?: number;
  maxSalary?: number;
  skills?: string[];
  languages?: string[];
  location?: string;
}
